Content.Shared/RCD/Systems/RCDSystem.cs

```diff
@@ -35,313 +41,590 @@ public sealed class RCDSystem : EntitySystem
     [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
     [Dependency] private readonly SharedInteractionSystem _interaction = default!;
     [Dependency] private readonly SharedPopupSystem _popup = default!;
-    [Dependency] private readonly TagSystem _tag = default!;
     [Dependency] private readonly TurfSystem _turf = default!;
     [Dependency] private readonly IGameTiming _gameTiming = default!;
-
-    private readonly int RcdModeCount = Enum.GetValues(typeof(RcdMode)).Length;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly IPrototypeManager _protoManager = default!;
+    [Dependency] private readonly SharedMapSystem _mapSystem = default!;
 
     public override void Initialize()
     {
         base.Initialize();
 
+        SubscribeLocalEvent<RCDComponent, ComponentInit>(OnInit);
         SubscribeLocalEvent<RCDComponent, ExaminedEvent>(OnExamine);
-        SubscribeLocalEvent<RCDComponent, UseInHandEvent>(OnUseInHand);
         SubscribeLocalEvent<RCDComponent, AfterInteractEvent>(OnAfterInteract);
         SubscribeLocalEvent<RCDComponent, RCDDoAfterEvent>(OnDoAfter);
         SubscribeLocalEvent<RCDComponent, DoAfterAttemptEvent<RCDDoAfterEvent>>(OnDoAfterAttempt);
+        SubscribeLocalEvent<RCDComponent, RCDSystemMessage>(OnRCDSystemMessage);
+        SubscribeNetworkEvent<RCDConstructionGhostRotationEvent>(OnRCDconstructionGhostRotationEvent);
     }
 
-    private void OnExamine(EntityUid uid, RCDComponent comp, ExaminedEvent args)
+    #region Event handling
+
+    private void OnInit(EntityUid uid, RCDComponent component, ComponentInit args)
     {
-        if (!args.IsInDetailsRange)
+        // On init, set the RCD to its first available recipe
+        foreach (var protoId in component.AvailablePrototypes)
+        {
+            var proto = _protoManager.Index(protoId);
+
+            if (proto != null)
+            {
```

Comment:
Index can never return null, also you should be doing this on MapInit so saving doesn't bulldoze it.